ublk.cpp

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: MIT or GPL-2.0-only
+
+#include "config.h"
+#include "ublksrv_tgt.h"
+
+static int ublksrv_execve_helper(const char *op, const char *type, int argc, char *argv[])
+{
+	char *cmd, **nargv;
+	char *nenv[] = { NULL };
+	int i;
+
+	asprintf(&cmd, "ublk.%s", type);
+	nargv = (char **)calloc(argc + 2, sizeof(char *));
+	if (!nargv)
+		return -ENOMEM;
+	nargv[0] = cmd;
+	nargv[1] = (char *)op;
+	for (i = 1; i < argc; i++)
+		nargv[i + 1] = argv[i];
+
+	return execve(nargv[0], nargv, nenv);
```

Comment:
Here the whole path of ublk need to be taken into account, such as:

cd tests

/$HOME/git/ublksrv/tests/../ublk add -t null

nothing is dumped, and $? is 255, and this way actually fails all built test, such as:

make test T=generic/003